platform/mellanox/non-upstream-patches/README.md

```diff
@@ -0,0 +1,25 @@
+## Mellanox non-upstream linux kernel patches ##
+
+To include non-upstream patches into the sonic-linux image during build time, this folder must contain a patch archive.
+
+### Structure of the patch archive
+
+  1. The tarball should be gzip compressed i.e. of extension .tar.gz

```

Comment:
Why hard code the compression algorithm?

Reply:
For `tar`, the compression algorithm doesn't need to be specified to be able to extract the patches. I would instead rephrase this to specify various compression formats (.tar.gz, .tar.xz, .tar.bz2, etc.).